gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +164,233 @@ def _get_prior_measurements_in_world_frame(
+        inliers = set()
+        for outlier_weight_dict in outlier_weights:
+            for w_i1Ui2 in w_i1Ui2_measurements:
+                i1, i2 = w_i1Ui2.key1(), w_i1Ui2.key2()
+                outlier_weights_sum[(i1, i2)] += outlier_weight_dict[(i1, i2)]
+        for (i1, i2), weight_sum in outlier_weights_sum.items():
+            if weight_sum / len(projection_directions) < OUTLIER_WEIGHT_THRESHOLD:
+                inliers.add((i1, i2))
+
+        # Filter outliers, index back from symbol to int.
+        inlier_w_i2Ui1_dict = {}
+        inlier_w_i2Ui1_dict_tracks = {}
+        inlier_cameras: Set[int] = set()
+        for (i1, i2) in w_i2Ui1_dict:
+            if (C(i2), C(i1)) in inliers:  # there is a flip in indices from w_i2Ui1_dict to inliers.
+                inlier_w_i2Ui1_dict[(i1, i2)] = w_i2Ui1_dict[(i1, i2)]
+                inlier_cameras.add(i1)
+                inlier_cameras.add(i2)
+
+        for (track_id, cam_id) in w_i2Ui1_dict_tracks:
+            if (C(cam_id), L(track_id)) in inliers and cam_id in inlier_cameras:
+                inlier_w_i2Ui1_dict_tracks[(track_id, cam_id)] = w_i2Ui1_dict_tracks[(track_id, cam_id)]
+
+        return inlier_w_i2Ui1_dict, inlier_w_i2Ui1_dict_tracks, inlier_cameras
+
+    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]) -> gtsam.Values:
         """Converts translations from a list of absolute poses to gtsam.Values for initialization.
 
         Args:
-            wTi_initial: List of
+            wTi_initial: List of absolute poses.
+
+        Returns:
+            gtsam.Values containing initial translations (uses symbols for camera index).
         """
         initial = gtsam.Values()
         for i, wTi in enumerate(wTi_initial):
             if wTi is not None:
-                initial.insertPoint3(i, wTi.value.translation())
+                initial.insertPoint3(C(i), wTi.value.translation())
         return initial
 
+    def _select_tracks_for_averaging(
+        self,
+        tracks: List[SfmTrack2d],
+        valid_cameras: Set[int],
+        intrinsics: List[Optional[gtsfm_types.CALIBRATION_TYPE]],
+        tracks_to_cameras_ratio: float = TRACKS_TO_CAMERAS_RATIO,
+    ) -> List[SfmTrack2d]:
+        """Removes bad tracks and returns the longest ones from the rest based on tracks_to_cameras_ratio.
+
+        Bad tracks are those that have fewer than 3 measurements from valid_cameras.
+        Sorts the remaining tracks in descending order by number of measurements, and returns as many tracks
+        as tracks_to_cameras_ratio * number of valid cameras.
+
+        Args:
+            tracks: List of all input tracks.
+            valid_cameras: Set of valid camera indices (these have direction measurements and valid rotations).
+            intrinsics: List of camera intrinsics.
+            tracks_to_cameras_ratio: Ratio of tracks to cameras to use for averaging.
```

Comment:
What happens if a camera does have X number of tracks? does it mean every camera needs to have N tracks?

Reply:
I think i'm misunderstanding the variable name since we use it below as (# cameras) * ratio. Maybe we could clarify that this var just means that we will use X times as many tracks as valid cameras exist.

Reply:
Got it, maybe if we amend this slightly to 

```python
tracks_to_cameras_ratio: Ratio of total number of tracks to use to total number of cameras.
```